tests/lms/views/predicates/_lti_launch_test.py

```diff
@@ -0,0 +1,165 @@
+from unittest import mock
+
+from pyramid.testing import DummyRequest
+import pytest
+
+from lms.models import ModuleItemConfiguration
+from lms.views.predicates import (
+    DBConfigured,
+    CanvasFile,
+    URLConfigured,
+    Configured,
+    AuthorizedToConfigureAssignments,
+)
+
+
+class TestDBConfigured:
+    @pytest.mark.parametrize("value,expected", [(True, True), (False, False)])
+    def test_when_theres_a_matching_assignment_config_in_the_db(
+        self, pyramid_request, value, expected
+    ):
+        pyramid_request.params = {
+            "resource_link_id": "test_resource_link_id",
+            "tool_consumer_instance_guid": "test_tool_consumer_instance_guid",
+        }
+        predicate = DBConfigured(value, mock.sentinel.config)
```

Comment:
Since all the predicate construction/calls have the same structure and there are only two possibilities for whether the predicate value and result match, I was imagining something like:

```py
class TestDBConfigured:
  def test_when_theres_a_matching_assignment_config_in_the_db(self, pyramid_request):
     ...
     _check_predicate(DBConfigured, pyramid_request, should_match=True)

def _check_predicate(predicate_class, request, should_match):
  true_predicate = predicate_class(True, mock.sentinel.config)
  assert true_predicate(mock.sentinel.context, request) is should_match

  false_predicate = predicate_class(False, mock.sentinel.config)
  assert false_predicate(mock.sentinel.context, request) is not should_match
```

My rationale here is that, with less code, there is less likelihood of an unintended missing test case (eg. only checking one of the two cases for a predicate or repeating the same case twice in the parameter list).